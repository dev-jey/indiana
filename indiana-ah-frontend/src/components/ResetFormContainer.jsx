import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Circle } from 'better-react-spinkit';
import { withRouter } from 'react-router-dom';
import { sendResetLink } from '../redux/actions/authActions';
import InputField from './common/input/InputComponent.jsx';
import Button from '../styles/styledComponents/Button.jsx';
import { logo } from '../assets/images/svg';

export class ResetFormContainer extends Component {
  state = {
    email: ''
  };

  handleSubmit = async (event) => {
    event.preventDefault();
    await this.props.sendResetLink({ email: this.state.email }, this.props);
  };

  handleChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  };

  render() {
    const { email } = this.state;
    const { auth: { isLoading } } = this.props;
    return (
      <div className='signup-form-container'>
        <h3 className='text-center mt-5'>
          <img src={logo} alt='facebook logo' className='signup-title' />
          <hr />
        </h3>
        <form onSubmit={this.handleSubmit} className='signup-form mb-5'>
          <InputField
            width='100%'
            placeholder='Email'
            type='email'
            id='email'
            name='email'
            value={email}
            handleChange={this.handleChange}
          />
          <p className='text-center'>A reset link would be sent to your email</p>
          <Button
            type='submit'
            bgColor
            width={'100%'}
            margin={'auto'}
            height={'4.8rem'}
            sm>
            Reset Password
            {isLoading && (
              <span className='button-loading'>
                <Circle color={'rgba(255,255,255,1)'} />
              </span>
            )}
          </Button>
        </form>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps,
  { sendResetLink }
)(withRouter(ResetFormContainer));

ResetFormContainer.propTypes = {
  auth: PropTypes.object,
  displayForm: PropTypes.func,
  sendResetLink: PropTypes.func
};
