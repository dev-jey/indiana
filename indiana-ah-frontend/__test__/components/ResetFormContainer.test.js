import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ResetFormContainer } from '../../src/components/ResetFormContainer.jsx';


const props = {
  auth: {
    isLoading: false,
    error: '',
    sendResetLink: ''
  }
};

const props2 = {
  auth: {
    isLoading: true,
    error: '',
    sendResetLink: ''
  }
};

describe('Reset Form', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<ResetFormContainer {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('test the login container form', () => {
  const sendResetLink = jest.fn();
  const handleChange = jest.fn();
  const displayForm = jest.fn();
  it('should simulate the handleSubmit', () => {
    const wrapper = mount(
      <ResetFormContainer
        {...props}
        sendResetLink={sendResetLink}
        displayForm={displayForm}
        handleChange={handleChange}
      />
    );
    wrapper.find('form').simulate('submit');
    const input1 = wrapper.find('input').at(0);
    input1.simulate('change', { target: { value: 'fafa@gmail.com' } });
    expect(sendResetLink).toHaveBeenCalled();
  });
  it('should simulate the loading spinner', () => {
    const wrapper = mount(
      <ResetFormContainer
        {...props2}
        sendResetLink={sendResetLink}
        displayForm={displayForm}
        handleChange={handleChange}
      />
    );
    const span = wrapper.find('span').at(0);
    expect(span).toBeDefined();
  });
});
