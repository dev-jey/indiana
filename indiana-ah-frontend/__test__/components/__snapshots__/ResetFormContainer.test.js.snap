// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Form should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.8rem;
  margin: 0.1rem auto;
  height: 4.3rem;
  fill: #fbfcff;
  border: 0.05rem solid rgba(0,0,0,0.4);
  border-radius: 0.4rem;
  width: 100%;
  box-sizing: border-box;
  padding: 0.8rem 1.9rem;
  height: 4.3rem;
  background: #ffffff;
  border: 0.05rem solid rgba(0,0,0,0.1);
  border-radius: 0.2rem;
}

.c1 {
  background: #0B41CD;
  font-size: 2rem;
  line-height: 2.7rem;
  align: Center;
  -webkit-letterspacing: 5%;
  -moz-letterspacing: 5%;
  -ms-letterspacing: 5%;
  letterspacing: 5%;
  margin: auto;
  height: 4.8rem;
  border-radius: 0.4rem;
  text-align: center;
  color: #fff;
  cursor: pointer;
  border: none;
  text-transform: capitalize;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  width: 100%;
  padding: .4rem 2.5rem;
  display: block;
}

.c1:disabled {
  color: #dddddd;
  border: 0.2rem solid #dddddd;
}

@media (max-width:76.8rem) {
  .c0 {
    width: 100%;
    padding: 0.8rem;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    height: 3.5rem;
    font-size: 1.5rem;
  }
}

@media (max-width:830px) {
  .c1 {
    width: auto;
    line-height: 2rem;
    font-size: 1.7rem;
  }
}

@media (max-width:355px) {
  .c1 {
    width: auto;
    line-height: 1.5rem;
    padding: .4rem 1.5rem;
  }
}

<div
  className="signup-form-container"
>
  <h3
    className="text-center mt-5"
  >
    <img
      alt="facebook logo"
      className="signup-title"
      src="test-file-stub"
    />
    <hr />
  </h3>
  <form
    className="signup-form mb-5"
    onSubmit={[Function]}
  >
    <div
      className="input-field"
    >
      <label
        htmlFor="email"
      />
      <input
        className="c0"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
      <small
        className="text-danger ml-3 form-text"
      />
    </div>
    <p
      className="text-center"
    >
      A reset link would be sent to your email
    </p>
    <button
      className="c1"
      height="4.8rem"
      type="submit"
      width="100%"
    >
      Reset Password
    </button>
  </form>
</div>
`;
